### ISSUE TITLE
Webserver Review Problem: Data salad after displaying the HTML source code

### ISSUE THREAD

**Issue by jooseef** [mnemosyne-proj/mnemosyne#226]
Hello! o/
I discovered a problem that can be reproduced like this:
Show question from Card A, then show the answer and rate the answer.
Show question from card B. Show the HTML source of question B.
But the HTML source code shows the question of the next card C (should be question B of course).
Now show answer B. The answer is card C's question and card C's answer.
Card B is evaluated and removed from the not-memorized-stack even though the answer was not shown.

https://user-images.githubusercontent.com/108878285/186001618-22a6f28f-b949-4ffd-9eb4-d7ea46208532.mp4

The reason could be: The web server "thinks" that card B should be evaluated after form = cgi.FieldStorage(fp=environ["wsgi.input"], environ=environ) was called. I have no idea how to quickly fix the problem. Unfortunately I'm not familiar with the code structures of Mnemosyne.

**Comment by pbienst** [comment]
Strange..  Is this reproducible? Does it work fine if you do not display the html source?

**Comment by jooseef** [comment]
Yes I could reproduce it. I created these 3 test-cards as described above. It works fine if I don't display the source, yes.

**Comment by pbienst** [comment]
In that case, I suspect that the 'show source' functionality performs some a spurious extra call to the webserver, which messes up its internal logic. Given that average users are unlikely to run into this, I'm not sure we should spend time figuring this one out...
